core/chains/evm/client/node.go

```diff
@@ -353,6 +376,16 @@ func (n *node) cancelInflightRequests() {
 	n.chStopInFlight = make(chan struct{})
 }
 
+// unsubscribeAll unsubscribes all subscriptions
+// WARNING: NOT THREAD-SAFE
+// This must be called from within the n.stateMu lock
+func (n *node) unsubscribeAll() {
+	for _, sub := range n.subs {
+		sub.Unsubscribe()
+	}
+	n.subs = make([]ethereum.Subscription, 0)
```

Comment:
`make(T, 0)` seems pretty irregular. IMO setting nil here (and leaving unitialized to nil in the constructor) is simpler, but otherwise a literal empty slice would be more idiomatic:
```suggestion
	n.subs = []ethereum.Subscription{}
```